### PR TITLE
sql/sqlcheck: move analyzers to be driver-specific

### PR DESCRIPTION
**Pull request by a8m**
Also, detect modify tables changes and ignore report them as destructive. See examples for before/after:

image

image

### CODE REVIEW

File: sql/mysql/mysqlcheck/mysqlcheck.go [hunk 1]
```diff
@@ -15,14 +15,15 @@ import (
...
 type (
-	// DataDependOptions defines the configuration options
-	// for the data-dependent changes checker.
+	// DataDependOptions defines the additional configuration
+	// options for the data-dependent changes checker.
 	DataDependOptions struct {
-		// NotNull indicates if the analyzer should check for modification or
-		// addition of not null constraints to
+		// NotNull indicates if the analyzer should check for modification
+		// or addition of not null constraints to columns.
```
Comment (masseelch) [inline sql/mysql/mysqlcheck/mysqlcheck.go:26]:
```suggestion
		// or addition of NOT NULL constraints to columns.
```

File: sql/mysql/mysqlcheck/mysqlcheck.go [hunk 2]
```diff
@@ -15,14 +15,15 @@ import (
...
 		NotNull *bool `spec:"not_null,omitempty"`
```
Comment (masseelch) [inline sql/mysql/mysqlcheck/mysqlcheck.go:27]: Why is a pointer needed here?

File: sql/mysql/mysqlcheck/mysqlcheck.go [hunk 3]
```diff
@@ -188,3 +182,16 @@ func (d *DataDepend) addColumn(p *sqlcheck.Pass, change *sqlcheck.Change, t *sch
...
+	sqlcheck.Register(mysql.DriverName, func(r *schemahcl.Resource) (sqlcheck.Analyzer, error) {
+		az, err := NewDataDepend(r)
+		if err != nil {
+			return nil, err
+		}
+		return sqlcheck.Analyzers{
+			destructive.New(destructive.Options{}),
+			az,
+		}, nil
+	})
```
Comment (masseelch) [inline sql/mysql/mysqlcheck/mysqlcheck.go:196]: This means runtime errors instead of compile time, right?

**Review by masseelch — APPROVED** [review]

File: sql/mysql/mysqlcheck/mysqlcheck.go (hunk 3 above)
Comment (a8m) [reply sql/mysql/mysqlcheck/mysqlcheck.go:196]: Both happen on initializations because an error is returned in the case of an invalid configuration.

**Pull request merged**